packages/@webex/internal-plugin-metrics/test/unit/spec/call-diagnostic/call-diagnostic-metrics-latencies.ts

```diff
@@ -111,6 +128,62 @@ describe('internal-plugin-metrics', () => {
       assert.deepEqual(cdl.getMeetingInfoReqResp(), 10);
     });
 
+    describe('measureLatency', () => {
+      let clock;
+      let callbackStub;
+      let saveLatencySpy;
+
+      beforeEach(() => {       
+        clock = sinon.useFakeTimers(); 
+        
+        saveLatencySpy = sinon.stub(cdl, 'saveLatency');
+      });
+
+      afterEach(() => {
+        clock.restore();
+        sinon.restore();
+      });
+      
+      it('checks measureLatency with overwrite false', async () => {
+        const key = 'internal.client.pageJMT';
+        const overwrite = false;
+        callbackStub = sinon.stub().resolves();
+
+        const promise = cdl.measureLatency(callbackStub, 'internal.client.pageJMT', overwrite);
+        clock.tick(50);
```

Comment:
suggestion: Whilst this does happen to work, it is entirely reliant on the fact that nothing else runs between creating the promise, and incrementing the clock. If an `await` is added between these, the test will fail since the clock won't have incremented soon enough.

To avoid this, my suggestion is to have a custom stub implementation which increments the time when called. So you would end up defining the callback as something like:

```js
const callback = sinon.stub().callsFake(() => {
  clock.tick(50);
  return Promise.resolve(resolution);
});
```

This way, the clock is incremented as a part of being called, as though it took some time to run. The `clock.tick` here is no longer needed of course.